File: simple-fun-number-290-sum-of-threes/simple-fun-number-290-sum-of-threes.js
function sumOfThrees(n) {
  let m  = n.toString(3)
  return /^[10]+$/.test(m) ? m.split("").reduce((a,b,i)=>b!=="0"?`${a}+${3}^${m.length-i-1}`: a,"").replace(/^\+/,"")  : "Impossible"
}

File: simple-fun-number-290-sum-of-threes/simple-fun-number-290-sum-of-threes.py
def sum_of_threes(n):
    try:
        s = []
        t =  ternary(n)
        if t.count("2"):
            return "Impossible"
        for i,j in enumerate(t[::-1]):
            if j =="1":
                s.append("3^"+str(i))
        return "+".join(s[::-1])
    except:
        return "Impossible"
        
def ternary(n,b=3):
    e = n//b
    q = n%b
    if n == 0:
        return '0'
    elif e == 0:
        return str(q)
    else:
        return ternary(e, b) + str(q)